OpenRPA.NativeMessagingHost/myAddon2/background.js

```diff
@@ -974,3 +974,64 @@ async function downloadsOnChanged(delta) {
     if (port != null) port.postMessage(JSON.parse(JSON.stringify(message)));
 
 }
+
+async function getFocusedTab() {
+    let [tab] = await tabsquery({ active: true, lastFocusedWindow: true });
+    return tab;
+}
+
+function getCurrentBrowser() {
+    if (isChrome) return "chrome";
+    if (isFirefox) return "ff";
+    if (isChromeEdge) return "edge";
+}
+
+function isSameUrlOrigin(data) {
+    if (data.target && data.source) {
+        var target = new URL(data.target);
+        var source = new URL(data.source);
+
+        if (target.origin == source.origin)
+            return true;
+    }
+}
+
+function pushNavigateEvent(functionName, data, event) {
+    var message = {
+        functionName: functionName,
+        browser: data.browser,
+        windowId: data.tab.windowId,
+        tabid: data.tab.id,
+        tab: data.tab,
+        data: JSON.stringify({ source: data.source, target: data.target }),
+        result: JSON.stringify(event)
+    };
+
+    console.debug(`[send] ${message.functionName}`);
+
+    if(port!=null) port.postMessage(JSON.parse(JSON.stringify(message)));
+}
+
+async function handleOnBeforeNavigation(event) {
+    if (event.frameType == 'outermost_frame') {
+        var tab = await getFocusedTab();
+        var browser = getCurrentBrowser();
+        var data = {
+            browser: browser,
+            tab: tab,
+            target: tab.pendingUrl,
+            source: tab.url
+        }
+        
+        if (!isSameUrlOrigin(data))
```

Comment:
Is this required? I think we should trigger the event even if it is from the same origin

Reply:
Really, it is no required. I was asked to put this check validation, as a strategy to restrict events click that promote  navigations. And using this proposed solution approach it can happen when user click in some links that promotes navigation event. Although I found another way to restrict this scenario but I have some questions that We need discuss with the team.

Reply:
This `isSameUrlOrigin` as check validation was removed. And the event `chrome.OnBeforeNavidation` was replaced by `chrome.OnCommittedNavigation` it will improve ability to filter the transitions types of navigation event: type or auto bookmark.